### PR TITLE
Fixed MDS-1170

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/20187

Fixes https://arangodb.atlassian.net/browse/MDS-1170

* Fixed MDS-1170: Significant performance degradation when queries are executed within transactions that involve edits.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.11: this PR
  - [x] Backport for 3.10: https://github.com/arangodb/arangodb/pull/20189

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/MDS-1170
- [ ] Design document:

### CODE REVIEW

File: js/common/bootstrap/errors.js [hunk 1]
```diff
@@ -340,6 +340,7 @@
     "ERROR_SUPERVISION_GENERAL_FAILURE" : { "code" : 20501, "message" : "general supervision failure" },
     "ERROR_QUEUE_FULL"             : { "code" : 21003, "message" : "queue is full" },
     "ERROR_QUEUE_TIME_REQUIREMENT_VIOLATED" : { "code" : 21004, "message" : "queue time violated" },
+    "ERROR_TOO_MANY_DETACHED_THREADS" : { "code" : 21005, "message" : "too many detached scheduler threads" },
```
Comment (jsteemann) [inline js/common/bootstrap/errors.js:343]: note: this change does not really belong to this PR, but a previous PR changed `lib/Basics/errors.dat` without adding the changed generated JavaScript file `errors.js`. 
My PR now does that, because otherwise one will always end up with a modified local file after building 3.11.

**Review by MBkkt — APPROVED** [review]

File: CHANGELOG [hunk 2]
```diff
@@ -19,11 +22,6 @@ v3.11.6 (XXXX-XX-XX)
   connection statistics were built up in memory, but were never released because
   the statistics background thread was not running.
 
-* Avoid memory leak in case an arangod instance is started with the option
```
Comment (neunhoef) [inline CHANGELOG:22]: This does not look right, should the entry really be removed?

**Review by neunhoef — CHANGES_REQUESTED** [review]
LGTM, but see CHANGELOG.

**Review by neunhoef — APPROVED** [review]
LGTM
